ultralist/date_parser.go

```diff
@@ -0,0 +1,111 @@
+package ultralist
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// DateParser is a thing that parses relative, arbitrary, and absolute dates, and returns a date in the format of yyyy-mm-dd
+type DateParser struct{}
+
```

Comment:
a nice to have (and something I was toying with in my head) was to have a

var InteralDateFormat = "2006-01-02", so we can replace all the "2006-01-02" thoguhout the code.

Or maybe updating the formats to use RFC3339 ("2006-01-02T15:04:05Z07:00"), that way if we wanted to, we can have 'due tomorrow by 1pm'

Reply:
I think using a const for the internal date format is a good idea.

However, I'm against introducing the concept of hours/minutes into ultralist because I think a calendar or reminders app is a better tool for that job.

Reply:
sure thing.

Reply:
add a `DATE_FILTER` const to `app.go`.